sync/src/main/java/eu/metawired/wtrocki/sync/SyncSdk.java

```diff
@@ -0,0 +1,32 @@
+package eu.metawired.wtrocki.sync;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.aerogear.mobile.core.api.Core;
+import org.aerogear.mobile.core.api.SdkCore;
+import org.aerogear.mobile.core.configuration.ServiceConfig;
+
+/**
+ * Sync service SDK
+ *
+ * Need to be used in top level app:
+ *
+ * new SyncSdk(context).performSync();
+ */
+public class SyncSdk {
+
+    private Core core;
+
+    public SyncSdk(Context context) {
+        this.core = new SdkCore(context);
```

Comment:
So does this mean that each ServiceModule has its own instance of core?  If so I don't really like that because this means that for every service module a user is using they are also making new instances of the thread pools, network resources, etc.  While it probably won't be a problem in practice, I think that SDK Core should be a shared resource.

Reply:
I am pretty sure we can make the core singleton. Another option is to instantiate Core once and pass it to each of the service module.

Reply:
There are two variants of this. One is using constructor DI second one initializes core inside. I pointed out both and agree that constructor based aproach will be better way to go (config and some initialization is happening once. Problem with that is that network based layer will need to be shared with the services, so that's something I would like to talk about in terms of interceptors etc. 

TL;DR - service initialization means service specific seed data, 
Global initialization for core will mean some global handlers.

Reply:
Is there a way to avoid the developer having to manually inject a Core instance into each SDK they want to use?
Looking at Firebase as an example https://firebase.google.com/docs/database/android/start/ it seems you only need to:
- include the json defition file
- firebase dependencies

then you can do this:
```java
FirebaseDatabase database = FirebaseDatabase.getInstance();
DatabaseReference myRef = database.getReference("message");

myRef.setValue("Hello, World!");
```

That seems like a nice goal for our SDK. The develop only needs the mobile-config.json and the SDK dependencies, then they can get an instance of whatever SDK they want and use it.

Reply:
I think we need some clearing up on how services depending on others that are not the Core would look like. It wouldn't be safe to assume that we won't ever have dependencies between them.

We might need to not cater to diamond-style dependencies and avoid those ourselves, though.

Reply:
@david-martin @paolobueno I added some thoughts here : https://github.com/aerogear/aerogear-android-sdk/pull/10#issuecomment-358636869

Reply:
The whole idea is to not initialize services in core - just to have core injected.
Having core singleton is possible and definitely way to go.

>  It wouldn't be safe to assume that we won't ever have dependencies between them.

That will be huge problem and general anti-pattern.

Reply:
"everything's a singleton" seems very nice too. Considering caching done for commonjs modules I think this would become very idiomatic for the cordova sdk too.